openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py

```diff
@@ -400,8 +400,12 @@ def fill_account_move_line_currency_id(env):
         UPDATE account_move_line
         SET currency_id = company_currency_id
         WHERE currency_id IS NULL
+        RETURNING id
         """,
     )
+    ids = [x[0] for x in env.cr.fetchall()]
+    if ids:
+        env['account.move.line'].browse(ids)._compute_amount_residual()
```

Comment:
I thought that orm function was not available in the pre-migration script.
Or did I missed something ?

Reply:
You are correct, all account models are not in the environment during pre-migration.

Reply:
thanks for your review.

hum don't know how to implement the @remi-filament needs.

sql request should be done in pre-migration script.
call to _compute_amount_residual should be done on post-migration script but only on lines impacted by the request of the pre-migration script.
we could add a temporary extra column to_recompute_amount_residual, but it seems a little bit tricky. Do you have another idea ?